presto-hudi/src/main/java/com/facebook/presto/hudi/split/HudiPartitionSplitGenerator.java

```diff
@@ -0,0 +1,167 @@
+
+    public HudiPartitionSplitGenerator(
+            ConnectorSession session,
+            ExtendedHiveMetastore metastore,
+            HudiTableLayoutHandle layout,
+            HoodieTableFileSystemView fsView,
+            AsyncQueue<ConnectorSplit> asyncQueue,
+            ArrayDeque<String> partitionQueue,
+            String latestInstant)
+    {
+        this.metastore = requireNonNull(metastore, "metastore is null");
+        this.metastoreContext = toMetastoreContext(requireNonNull(session, "session is null"));
+        this.layout = requireNonNull(layout, "layout is null");
+        this.table = layout.getTable();
+        this.tablePath = new Path(table.getPath());
+        this.fsView = requireNonNull(fsView, "fsView is null");
+        this.asyncQueue = requireNonNull(asyncQueue, "asyncQueue is null");
+        this.partitionQueue = requireNonNull(partitionQueue, "partitionQueue is null");
+        this.latestInstant = requireNonNull(latestInstant, "latestInstant is null");
+        this.splitWeightProvider = createSplitWeightProvider(requireNonNull(session, "session is null"));
+        this.isRunning = true;
+    }
+
+    @Override
+    public void run()
+    {
+        HoodieTimer timer = new HoodieTimer().startTimer();
+
+        while (isRunning || !partitionQueue.isEmpty()) {
+            String partitionName = null;
+            synchronized (partitionQueue) {
+                if (!partitionQueue.isEmpty()) {
+                    partitionName = partitionQueue.pollFirst();
+                }
+            }
+
+            if (partitionName != null) {
+                generateSplitsFromPartition(partitionName);
+            }
+        }
+        LOG.debug(String.format("HudiPartitionSplitGenerator %s finishes in %d ms", this, timer.endTimer()));
+    }
+
+    public void stopRunning()
+    {
+        this.isRunning = false;
+    }
+
+    private void generateSplitsFromPartition(String partitionName)
+    {
+        HudiPartition hudiPartition = getHudiPartition(metastore, metastoreContext, layout, partitionName);
+        Path partitionPath = new Path(hudiPartition.getStorage().getLocation());
+        String relativePartitionPath = FSUtils.getRelativePartitionPath(tablePath, partitionPath);
+        fsView.getLatestFileSlicesBeforeOrOn(relativePartitionPath, latestInstant, false)
+                .map(fileSlice -> createHudiSplit(table, fileSlice, latestInstant, hudiPartition, splitWeightProvider))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .forEach(asyncQueue::offer);
+    }
+
+    private Optional<HudiSplit> createHudiSplit(
+            HudiTableHandle table,
+            FileSlice slice,
+            String timestamp,
+            HudiPartition partition,
+            HudiSplitWeightProvider splitWeightProvider)
+    {
+        HudiFile hudiFile = slice.getBaseFile().map(f -> new HudiFile(f.getPath(), 0, f.getFileLen())).orElse(null);
```

Comment:
base file can be null in certain cases when Hudi allows inserts to go to log files.

Reply:
Yes.  This logic of calculating bytes is actually moved from `HudiSplitManager`.  The `map` function only applies when the option is non-empty.